File: DarkOrbit 10.0/Net/netty/commands/FactionModule.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Ow.Utils;

namespace Ow.Net.netty.commands
{
    class FactionModule : command_i3O
    {
        public const short ID = 2289;

        public const short MMO = 1;
        public const short VRU = 3;
        public const short EIC = 2;
        public const short NONE = 0;

        public short factionId = 0;

        public FactionModule(short factionId)
        {
            this.factionId = factionId;
        }

        public override byte[] write()
        {
            var param1 = new ByteArray(ID);
            super(param1);
            param1.writeShort(this.factionId);
            param1.writeShort(-31859);
            return param1.Message.ToArray();
        }
    }
}

File: DarkOrbit 9.0/DarkOrbit 9.0 Emulator/src/simulator/netty/serverCommands/FactionModule.java
package simulator.netty.serverCommands;

import java.io.DataOutputStream;
import java.io.IOException;

public class FactionModule
        extends class_418 {

    public static int ID = 2430;

    public short faction = NONE;

    public static short NONE = 0;
    public static short MMO  = 1;
    public static short VRU  = 3;
    public static short EIC  = 2;

    public FactionModule(short pFaction) {
        this.faction = pFaction;
    }

    public int getID() {
        return ID;
    }

    public int method_1005() {
        return 0;
    }

    public void write(DataOutputStream out) {
        try {
            out.writeShort(ID);
            this.writeInternal(out);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    protected void writeInternal(DataOutputStream out) {
        try {
            super.writeInternal(out);
            out.writeShort(2035);
            out.writeShort(this.faction);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}